tests/testthat/test-aes-calculated.r

```diff
@@ -49,3 +49,19 @@ test_that("make_labels() deprases mappings properly", {
   expect_identical(make_labels(aes(x = 1)), list(x = "x"))
   expect_identical(make_labels(aes(x = NULL)), list(x = "x"))
 })
+
+test_that("staged aesthetics warn appropriately for duplicated names", {
+  # Test should *not* report `NA` as the duplicated aes (#4707)
+  df <- data.frame(x = 1, y = 1, lab = "test")
+  expr <- substitute(
```

Comment:
Since I don't think you're actually doing any substitution, `quote()` would be more appropriate here. But I think you could also just do `p <- ggplot() + ...` in the `expect_snapshot_warning()`.

Reply:
Good points, I wasn't aware of this, so thanks!